### ISSUE TITLE
[Bug?] undefined is converted to null

### ISSUE THREAD

**Issue by dotvirus** [eemeli/yaml#173]
#### Version
Running on 1.10.0

#### Reproduce

```js
const YAML = require("yaml");

const obj = () => ({
  a: 4,
  b: "str",
  c: null,
  d: undefined,
});

const jsonified = JSON.parse(JSON.stringify(obj()));
console.log(jsonified.hasOwnProperty("d")); // -> false

const yamlified = YAML.parse(YAML.stringify(obj()));
console.log(yamlified.hasOwnProperty("d")); // -> true???
```

I don't see how this would be intentional. undefined is not equal to null.

**Comment by eemeli** [comment]
This was a design choice of mapping `undefined` to `null` in YAML, as it's the closest match to an empty value. But I do see your point that this is different from what `JSON.stringify()` does.

I think this should be considered in terms of the 2.0 update, as changing the API like this can be a breaking change.

**Comment by kyeotic** [comment]
I would like to vote that undefined keys not remain in the stringified version. I am now manually preprocessing to remove these keys, deeply.

**Comment by eemeli** [comment]
Yeah, happy to include this fix in the major update. A PR on this would be welcome; I'm finding myself spending a bit more time on a couple of other projects atm.

**Comment by kyeotic** [comment]
@eemeli I tried to dig into this. The code is pretty complicated, but it looks like some information is lost when construction the node graph so that `undefined` is converted into `null` as the `Pair` is made. That means the change can't be isolated to stringify, it needs to be part of the document's construction.
